modules/formulize/class/durationElement.php

```diff
@@ -0,0 +1,271 @@
+			$ele_value['min_minutes'] = intval($_POST['min_minutes']);
+			$ele_value['max_minutes'] = intval($_POST['max_minutes']);
+
+			// Save input box size
+			$ele_value['size'] = intval($_POST['size']);
+
+			$element->setVar('ele_value', $ele_value);
+
+			return true;
+		}
+		return false;
+	}
+
+	// Convert form input to minutes for storage
+	function prepareDataForSaving($value, $element, $entry_id = null)
+	{
+		if (!is_array($value)) {
+			return NULL;
+		}
+
+		$ele_value = $element->getVar('ele_value');
+
+		$totalMinutes = 0;
+
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			if (isset($value[$unit]) && is_numeric($value[$unit])) {
+				$totalMinutes += $value[$unit] * $multiplier;
+			}
+		}
+
+		// Validate min/max duration
+		if ($ele_value['min_minutes'] > 0 && $totalMinutes < $ele_value['min_minutes']) {
+			return $ele_value['min_minutes'];
+		}
+		if ($ele_value['max_minutes'] > 0 && $totalMinutes > $ele_value['max_minutes']) {
+			return $ele_value['max_minutes'];
+		}
+
+		return $totalMinutes > 0 ? $totalMinutes : NULL;
+	}
+
+	function prepareDataForDataset($value, $handle, $entry_id) {
+		return $value;
+	}
+
+	// Convert stored minutes back to time units for display
+	function loadValue($value, $ele_value, $element)
+	{
+		if (!is_numeric($value)) {
+			return $ele_value;
+		}
+
+		$minutes = intval($value);
+		$breakdown = array();
+
+		// Convert minutes to time units
+		$remaining = $minutes;
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			if ($ele_value['show_' . $unit]) {
+				$amount = floor($remaining / $multiplier);
+				$breakdown[$unit] = $amount;
+				$remaining -= ($amount * $multiplier);
+			}
+		}
+
+		$ele_value['values'] = $breakdown;
+		return $ele_value;
+	}
+
+	// Render the duration inputs
+	function render($ele_value, $caption, $markupName, $isDisabled, $element, $entry_id, $screen = false, $owner = null)
+	{
+		if ($isDisabled) {
+			// Render as text for disabled state
+			$output = "";
+			if (isset($ele_value['values'])) {
+				foreach ($ele_value['values'] as $unit => $amount) {
+					if ($amount > 0 && $ele_value['show_' . $unit]) {
+						$output .= $amount . " " . $unit . " ";
+					}
+				}
+			}
+			return new XoopsFormLabel($caption, trim($output), $markupName);
+		}
+
+		// Create container for inputs
+		$container = new XoopsFormElementTray($caption, ' ');
+
+		// Add input field for each enabled time unit
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			$unitMarkupName = $markupName . '[' . $unit . ']';
+			if ($ele_value['show_' . $unit]) {
+				$value = isset($ele_value['values'][$unit]) ? $ele_value['values'][$unit] : '';
+				${"input_$unit"} = new XoopsFormText(
+					ucfirst($unit).":",
+					$unitMarkupName,
+					$ele_value['size'],
+					5,
+					$value,
+					false,
+					true
+				);
+				${"input_$unit"}->setExtra("min='0'");
+				${"input_$unit"}->setExtra("class='formulize-duration-element-input numbers-only-textbox'");
```

Comment:
I think you can leave off the numbers-only-textbox, because the TRUE as the last param of the textbox creation above, tells the system to do it as a numbers only box? Or did you find that wasn't effective by itself?